### ISSUE TITLE
remove clash

### ISSUE THREAD

**Issue by xbdeng** [NewbieOrange/SUSTech-SQL-Project2-Public#16]
If I remove a course, but there are already students selecting the course.

Should I give an error to forbid the remove or remove the course and the select course information together?

Similarly, other remove will meet this clash as well.

**Comment by JohnWestonNull** [comment]
We're looking into this matter, please stay tuned.
Do you have any suggestions?

**Comment by xbdeng** [comment]
I have some solutions, but all of them will lead to a great change in the interface. Because some groups have already completed this project, making too much change in the interface is not a good way.

Compromise, you could choose one or two remove functions to make a change, and the other removes just through an error and forbid the remove. This can both inspect our designing ability, and not make too much change.

If you want to make some change, I recommend that not modifying the function, but adding some(one or two may be good) 
 new functions to execute the truncate remove.

The remove function which is already exit through an error may be a good way. In reality, there is no rational reason to remove a course that already has students select or study. These messages should never be removed, or will lead to an avalanche truncate remove.

To sum up, either remove or error is not important. A clear description and unambiguous rule may be enough.

**Comment by ziqin** [comment]
The removal semantic in this system is defined as follows: To remove an entity from the system, related entities dependent on this entity (usually rows referencing the row to remove through foreign keys in a relational database) shall be removed together.
